contrib/cell_star/utils/image_util.py

```diff
@@ -0,0 +1,402 @@
+# -*- coding: utf-8 -*-
+__author__ = 'Adam Kaczmarek, Filip Mróz'
+
+# External imports
+from os.path import  exists
+import os
+from os import makedirs
+
+import numpy as np
+import scipy as sp
+import scipy.ndimage
+import scipy.misc
+import matplotlib.pyplot as plt
+from scipy.ndimage.filters import *
+from numpy import argwhere
+
+debug_image_path = ""
+
+SHOW = False
+SILENCE = True
+
+def prepare_debug_folder():
+    if not exists(debug_image_path):
+        makedirs(debug_image_path)
+
+def convolve2d(img, kernel, mode='same'):
+    return convolve(img, kernel)
+
+
+def extend_slices(my_slices, extension):
+    def extend_slice(my_slice, extension):
+        ind = (my_slice.indices(100000)[0] - extension, my_slice.indices(100000)[1] + extension)
+        return slice(*ind)
+    return (extend_slice(my_slices[0],extension), extend_slice(my_slices[1],extension))
+
+
+def get_bounding_box(image_mask):
+    """
+    Calculates the minimal bounding box for non zero elements.
+    @returns [ystart,ystop), [xstart,xstop) or None, None
+    """
+    non_zero_points = argwhere(image_mask)
+    if len(non_zero_points) == 0:
+        return None
+    (ystart, xstart), (ystop, xstop) = non_zero_points.min(0), non_zero_points.max(0) + 1
+    return (ystart, ystop), (xstart, xstop)
+
+
+def get_circle_kernel(radius):
```

Comment:
I have an instance where get_circle_kernel is being called with a value of 1.36. The kernel should be symmetric around the center point, but instead the value is:

    [[ False, False, False],
     [False, True, True],
     [False, True, True]]

The problem is that `np.ogrid[-radius:radius+epsilon]` returns the values, -1.36, -.36, .64, consistent with adding 1 to -1.36. The grid should instead be
 
    np.mgrid[np.floor(-kernel_size):np.ceil(kernel_size + .001) + 1,
        np.floor(-kernel_size):np.ceil(kernel_size + .001) + 1]